### ISSUE TITLE
Correct spacing not added when child elements are React Components

### ISSUE THREAD

**Issue by yuzeh** [akiran/react-slick#29]
Thanks for this library!

I tried initializing a `<Slider>` with a user defined React component as children, the proper spacing between elements to satisfy `slidesToShow` and `slidesToScroll` is not set. However, when I "inline" these elements, things work great.

Example code that doesn't work properly:
```javascript
  var Bar = React.createClass({
    render: function() {
      return (
          <div>
            <img src={this.props.imageUrl} />
          </div>
      );
    },
  });

  var Foo = React.createClass({
    render: function() {
      var items = this.props.items.map(function(item, i) {
        return (
          <Bar key={item.id} imageUrl={item.imageUrl} />
        );
      });

      var settings = {
        arrows: true,
        dots: true,
        responsive: false,
        infinite: false,
        draggable: false,
        slidesToShow: 10,
        slidesToScroll: 5,
      };

      return (
        <Slider className='my-class' {...settings}>
          {items}
        </Slider>
      )
    },
  });
```

Example code that works properly:
```javascript
  var Foo = React.createClass({
    render: function() {
      var items = this.props.items.map(function(item, i) {
        return (
          <div key={item.id}>
            <img src={item.imageUrl} />
          </div>
        );
      });

      var settings = {
        arrows: true,
        dots: true,
        responsive: false,
        infinite: false,
        draggable: false,
        slidesToShow: 10,
        slidesToScroll: 5,
      };

      return (
        <Slider className='my-class' {...settings}>
          {items}
        </Slider>
      )
    },
  });
```

Is this a known problem, and if so is there a good way to fix it?

**Comment by akiran** [comment]
Both should work.
Let me test it

**Comment by yuzeh** [comment]
So it looks like css `width` is not getting assigned to the items if the root element of the item is a React component.

Current workaround is wrapping the component in a `div`.

**Comment by akiran** [comment]
@yuzeh 

children of slider component to adjust  styles and class names based on intreractions
https://github.com/akiran/react-slick/blob/master/lib/inner-slider.jsx#L57-62

In this case, style and classnames are applied to react components instead of html elements.

Your work around is one way to fix this.

Other way would be to pass props to div inside component (I haven't tested it though)
```
var Bar = React.createClass({
    render: function() {
      return (
          <div {...this.props}>
            <img src={this.props.imageUrl} />
          </div>
      );
    },
  });
```

I will see if there is any better way to fix this issue.

**Comment by gus4no** [comment]
similar issue here but it's just a regular div, not a component

```  render: function() {
    var slickSettings = {
        dots: true,
        infinite: false,
        adaptiveHeight: false,
        speed: 500,
        slidesToShow: 7,
        slidesToScroll: 7,
        responsive: [{
          breakpoint: 1200,
            settings: {
              slidesToShow: 6,
              slidesToScroll: 6
            }
        }, {
          breakpoint: 1024,
            settings: {
              slidesToShow: 4,
              slidesToScroll: 4
            }
        }, {
          breakpoint: 768,
            settings: {
              slidesToShow: 3,
              slidesToScroll: 3
            }
        }, {
          breakpoint: 480,
            settings: {
              slidesToShow: 2,
              slidesToScroll: 2
            }
        }]
    };

    var subcategories = this.state.subcategories.map(function(subcategory){
      return(
          <div key={subcategory.id}>
            <div className="active-link-with-image tt-cp text-2">
              <a data-category={subcategory.slug}>
                <img src={subcategory.images.thumb}/>
                <div className="name">
                  <span>{subcategory.name}</span>
                </div>
              </a>
            </div>
          </div>
      )
    }.bind(this));
    
    return (
      <div className="slick-categories">
        <Slider {...slickSettings} ref="slick">
          {subcategories}
        </Slider>
        <div className="spinner loading" ref="spinner"/>
      </div>
    );
  }````

**Comment by akiran** [comment]
Slider is not getting update when new slides are arrived

Earlier we used to reinitialize slider when props changed, but removed it to fix other issue #85 
https://github.com/akiran/react-slick/commit/01e6e4b60adb61a1282bf358f77241f2e0efa861

Need to find a different solution

**Comment by mklb** [comment]
@yuzeh thanks for your fix... tried hours to get this working ;D
